provider/simple/provider.go

```diff
@@ -0,0 +1,72 @@
+// Package simple implements structures and methods to provide blocks,
+// keep track of which blocks are provided, and to allow those blocks to
+// be reprovided.
+package simple
+
+import (
+	"context"
+
+	cid "github.com/ipfs/go-cid"
+	q "github.com/ipfs/go-ipfs/provider/queue"
+	logging "github.com/ipfs/go-log"
+	routing "github.com/libp2p/go-libp2p-routing"
+)
+
+var logP = logging.Logger("provider.simple")
+
+const provideOutgoingWorkerLimit = 8
```

Comment:
It would be great to be able to set this from config (can be done in a separate PR)